### PR TITLE
Add entries viz share URL

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on: https://github.com/the-deep/server/pull/702

### CODE REVIEW

File: src/components/other/ProjectAddForm/index.js [hunk 1]
```diff
@@ -68,6 +68,8 @@ const projectAddSchema = {
     },
 };
 
+const shouldHidePrivate = ({ accessPrivateProject }) => !accessPrivateProject;
```
Comment (tnagorra) [inline src/components/other/ProjectAddForm/index.js]:
```suggestion
const shouldHidePrivateProject = ({ accessPrivateProject }) => !accessPrivateProject;
```

File: src/redux/initial-state/dev-lang.json [hunk 2]
```diff
@@ -1299,7 +1299,12 @@
         "-98549020": "Select Analysis Date",
         "-22196585": "Create a project",
         "-48980925": "Save and get started",
-        "-59904873": "Project details"
+        "-59904873": "Project details",
+        "-99774780": "The generated link can be publicly accessed by anyone without authentication and contains only non-confidential data.",
+        "-79488121": "Share",
+        "-99801867": "Share public link",
+        "-3100650": "Share public link",
```
Comment (tnagorra) [inline src/redux/initial-state/dev-lang.json]: But why do we have different entries for same text?

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — CHANGES_REQUESTED** [review]
Need some fixes!

File: src/views/Entries/ShareModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                    onShareLinkChange(response?.publicUrl);
+                }
+            },
+            onFailure: (_, errorBody) =>
+                notifyOnFailure(_ts('entries', 'shareVizLink'))({ error: errorBody }),
+        },
+    );
+
+    return (
+        <Modal
```
Comment (subinasr) [inline src/views/Entries/ShareModal/index.tsx:76]: Shouldn't we be using the new Modal component?

**Review by subinasr — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

File: src/views/Entries/ShareModal/index.tsx (hunk 3 above)
Comment (tnagorra) [reply src/views/Entries/ShareModal/index.tsx:76]: The new UI is not used on the Entries page. So, no!

**Pull request merged**